app/database/operator/server_data_operator/handlers/index.ts

```diff
@@ -53,11 +54,20 @@ export default class ServerDataOperatorBase extends BaseDataOperator {
             return [];
         }
 
+        const names = Array.from(new Set(emojis.map((e) => e.name)));
+
+        const locals = await queryCustomEmojisByName(this.database, names).fetch();
+
+        const deleteRawValues = locals.
+            filter((local) => !emojis.find((raw) => raw.id === local.id)).
+            map((local) => ({name: local.name}));
+
         return this.handleRecords({
             fieldName: 'name',
             transformer: transformCustomEmojiRecord,
             prepareRecordsOnly,
             createOrUpdateRawValues: getUniqueRawsBy({raws: emojis, key: 'name'}),
+            deleteRawValues,
```

Comment:
What is the impact on this on emojis that are currently rendered on the screen? And what about the re-renders of the components?

Reply:
Just to point out, that this will trigger a re-render, ofc, if the custom emoji have the id different, since the image will be replaced with the new custom emoji image.